parlai/tasks/self_chat/worlds.py

```diff
@@ -59,6 +146,22 @@ def parley(self):
                 }
                 self.acts[1 - i] = context
                 self.agents_ordered[i].observe(validate(context))
+            self.contexts = None
+        elif self.seed_utterances:
+            # grab the first two seed messages
+            utts = self.seed_utterances[:2]
+            # process the turn
+            for i in [0, 1]:
+                # if we have a seed utterance, add it to the conversation
+                if len(utts) > i:
```

Comment:
i'm a bit confused by this block. shouldn't we only get the seed_utterances if `self.turn_cnt == 0`? 

also, if we are always grabbing 2 seed utterances, shouldn't `len(utts)` always be greater than `i` if `i in [0, 1]`? what am i missing?

Reply:
Well, this is written so that there could be more or less than 2 seed utterances.

> shouldn't we only get the seed_utterances if `self.turn_cnt == 0`?

If there are also seed contexts, then the seed utterances will be added on the second turn, so `self.turn_cnt != 0`.

> if we are always grabbing 2 seed utterances, shouldn't len(utts) always be greater than i if i in [0, 1]?

We might only be grabbing a single seed utterance, since an odd number of seed utterances is supported. Let's say we want to start the conversation with just "Hey, what's up?" for instance.

I'll add some more comments to hopefully make this clearer. But let me know if there's something I'm missing as well!